connectors/kafka/src/main/java/forklift/producers/KafkaForkliftProducer.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    @Override
+    public String send(Map<Header, Object> headers, Map<String, String> properties, ForkliftMessage message)
+                    throws ProducerException {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public String send(Map<String, String> properties, ForkliftMessage message)
+                    throws ProducerException {
+        return this.sendForkliftWrappedMessage(message.getMsg(), properties);
+    }
+
+    @Override
+    public Map<String, String> getProperties() throws ProducerException {
+        return this.properties;
+    }
+
+    @Override
+    public void setProperties(Map<String, String> properties) throws ProducerException {
+        this.properties = properties;
+    }
+
+    @Override
+    public void setHeaders(Map<Header, Object> haeders) {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public Map<Header, Object> getHeaders() {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public void close() throws IOException {
+        //do nothing, the passed in KafkaProducer may be used elsewhere and should be closed by the KafkaController
+    }
+
+    private String sendForkliftWrappedMessage(String message, Map<String, String> messageProperties) throws ProducerException {
+        GenericRecord avroRecord = new GenericData.Record(forkliftSchema);
+        avroRecord.put(SCHEMA_FIELD_NAME_VALUE, message);
+        messageProperties = messageProperties == null ? new HashMap<>() : new HashMap<>(messageProperties);
+        //add the producer level properties but do not overwrite message level properties
+        for (Map.Entry<String, String> entry : this.properties.entrySet()) {
+            messageProperties.putIfAbsent(entry.getKey(), entry.getValue());
+        }
+        avroRecord.put(SCHEMA_FIELD_NAME_PROPERTIES, this.formatMap(messageProperties));
+        ProducerRecord record = new ProducerRecord<>(topic, null, avroRecord);
+        try {
+            RecordMetadata result = (RecordMetadata)kafkaProducer.send(record).get();
+            return result.topic() + "-" + result.partition() + "-" + result.offset();
+
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new ProducerException("Error sending Kafka Message", e);
+        } catch (ExecutionException e) {
+            throw new ProducerException("Error sending Kafka Message", e);
+        }
+    }
+
+    private Schema addForkliftPropertiesToSchema(Schema schema) throws IOException {
+        String originalJson = schema.toString(false);
+        JsonNode propertiesField = mapper.readTree("{\"name\":\"forkliftProperties\",\"type\":\"string\",\"default\":\"\"," +
```

Comment:
This inline JSON seems hard to read; can this at least be factored into a constant?